File: utils/institution.py
import os

DEFAULT_ESTABLISHMENT_NAME = "L.M. College of Pharmacy (LMCP)"
DEFAULT_ESTABLISHMENT_ADDRESS = (
    "L.M. College of Pharmacy, Navrangpura, Ahmedabad - 380009, Gujarat, India"
)
DEFAULT_CPCSEA_REGISTRATION_NUMBER = "228/PO/ReBi/S/2000/CPCSEA"
DEFAULT_CPCSEA_REGISTRATION_DATE = "16th June, 2000"
DEFAULT_ANIMAL_HOUSING_LOCATION = (
    "Institutional Animal House, L.M. College of Pharmacy, Ahmedabad - 380009"
)
DEFAULT_EXPERIMENT_LOCATION = (
    "L.M. College of Pharmacy, Ahmedabad - 380009 "
    f"(CPCSEA Reg. No. {DEFAULT_CPCSEA_REGISTRATION_NUMBER})"
)


def _env(name: str, default: str) -> str:
    return os.getenv(name, default).strip()


def get_establishment_name() -> str:
    return _env("LMCP_ESTABLISHMENT_NAME", DEFAULT_ESTABLISHMENT_NAME)


def get_establishment_address() -> str:
    return _env("LMCP_ESTABLISHMENT_ADDRESS", DEFAULT_ESTABLISHMENT_ADDRESS)


def get_cpcsea_registration_number() -> str:
    return _env("LMCP_CPCSEA_REGISTRATION_NUMBER", DEFAULT_CPCSEA_REGISTRATION_NUMBER)


def get_cpcsea_registration_date() -> str:
    return _env("LMCP_CPCSEA_REGISTRATION_DATE", DEFAULT_CPCSEA_REGISTRATION_DATE)


def get_animal_housing_location() -> str:
    return _env("LMCP_ANIMAL_HOUSING_LOCATION", DEFAULT_ANIMAL_HOUSING_LOCATION)


def get_experiment_location() -> str:
    reg = get_cpcsea_registration_number()
    default = (
        "L.M. College of Pharmacy, Ahmedabad - 380009 "
        f"(CPCSEA Reg. No. {reg})"
    )
    return _env("LMCP_EXPERIMENT_LOCATION", default)


def get_institutional_form_b_defaults() -> dict[str, str]:
    return {
        "establishment_name": get_establishment_name(),
        "establishment_address": get_establishment_address(),
        "registration_number": get_cpcsea_registration_number(),
        "registration_date": get_cpcsea_registration_date(),
        "animal_housing_location": get_animal_housing_location(),
        "experiment_location": get_experiment_location(),
    }

File: frontend/src/constants/institution.ts
export const INSTITUTIONAL_DEFAULTS = {
  establishmentName: "L. M. College of Pharmacy, Ahmedabad",
  establishmentAddress:
    "L.M. College of Pharmacy, Navrangpura, Ahmedabad - 380009, Gujarat, India",
  registrationNumber: "228/PO/ReBi/S/2000/CPCSEA",
  registrationDate: "16th June, 2000",
  animalHousingLocation:
    "Institutional Animal House, L.M. College of Pharmacy, Ahmedabad - 380009",
  experimentLocation:
    "L.M. College of Pharmacy, Ahmedabad - 380009 (CPCSEA Reg. No. 228/PO/ReBi/S/2000/CPCSEA)",
} as const;

export const RESEARCH_TYPES = [
  "Basic Research",
  "Educational",
  "Regulatory",
  "Contract Research",
] as const;

export const FORM_B_DECLARATIONS = [
  "declaration_not_duplicative",
  "declaration_qualified",
  "declaration_no_alternative",
  "declaration_iaec_approval_for_changes",
  "declaration_scientific_review",
  "declaration_hazardous_certificates",
  "declaration_form_d_records",
  "declaration_no_start_before_approval",
  "declaration_rehabilitation",
] as const;

export type FormBDeclarationKey = (typeof FORM_B_DECLARATIONS)[number];

export const FORM_B_DECLARATION_LABELS: Record<FormBDeclarationKey, string> = {
  declaration_not_duplicative:
    "I certify that the research proposal submitted is not unnecessarily duplicative of previously reported research.",
  declaration_qualified:
    "I certify that I am qualified and have experience in experimentation on animals.",
  declaration_no_alternative:
    "For painful procedures, I certify that I have reviewed pertinent scientific literature and found no valid alternative that may cause less pain or distress.",
  declaration_iaec_approval_for_changes:
    "I will obtain approval from the IAEC/CPCSEA before initiating any changes in this study.",
  declaration_scientific_review:
    "I certify that performance of the experiment will be initiated only upon review and approval of scientific intent by the appropriate expert body.",
  declaration_hazardous_certificates:
    "I certify that I will submit appropriate certification of review and concurrence for hazardous-agent studies.",
  declaration_form_d_records:
    "I shall maintain all records as per Form D and submit them to the IAEC.",
  declaration_no_start_before_approval:
    "I certify that I will not initiate the study before written IAEC/CPCSEA approval and will follow IAEC/CPCSEA recommendations.",
  declaration_rehabilitation:
    "I certify that rehabilitation policies will be adopted wherever required.",
};
